src/test/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectActionTest.java

```diff
@@ -20,6 +21,46 @@
 
 public class CourseProjectActionTest {
 
+  private class TestDialogs implements CourseProjectActionDialogs {
```

Comment:
does not it make test cases a bit dependent on this implementation? :octocat:

Reply:
Perhaps there is some confusion over the fact that, in this PR, `Dialogs` interface doesn't exist anymore but it is replaced with more specific-use interface `CourseProjectActionDialogs`. In the current design, if the old `Dialogs`-like interface are required elsewhere, there is nothing here that should be reused. This is only for `CourseProjectAction`, as the name suggests.

`CourseProjectActionDialogs`'s design is perhaps a bit sub-optimal because it's methods are in different levels of abstraction (compare `showOkCancelDialog` with `showImportIdeSettingsDialog`). One possibility would be keep old `Dialogs` interface and make `CourseProjectActionDialogs` an extension (subtype) to it. Similarly, `CourseProjectActionDialogsImpl` would inherit old `IntelliJDialogs`.

Reply:
Yes, very good point about `showImportIdeSettingsDialog` being on a different level of abstraction than the other more generic dialog methods.

I actually really like your idea about keeping the old `Dialogs` interface and making `CourseProjectActionDialogs` a subtype of it, as the `Dialogs` interface could be useful in other places as well.